extensions/sendbird/v1/actions/createUser/createUser.ts

```diff
@@ -84,3 +85,19 @@ export const createUser: Action<typeof fields, typeof settings> = {
     }
   },
 }
+
+const parseNickname = (payload: any): string => {
+  const nickname: string = payload.fields.nickname
+  if (!isEmpty(nickname)) return nickname
+
+  const firstName = payload.patient.profile?.first_name
+  const lastName = payload.patient.profile?.last_name
+
+  if (isEmpty(nickname) && (isEmpty(firstName) || isEmpty(lastName))) {
+    throw new Error(
+      'Nickname is not specified, and either the first name or last name is unknown'
+    )
+  }
```

Comment:
We shouldn't throw an error here, the order of logic is:
- use nickname if present
- use firstName lastName if both present
- if either firstName or lastName is not present then do not set nickname -- let the consumer (person configuring the extension) handle this

Reply:
I don't understand where the error should be raised. Because, from my perspective, it is reported in the correct function, unless you have an alternative plan for it?

Reply:
Hi @rahulkeerthi, according to [documentation](https://sendbird.com/docs/chat/platform-api/v3/user/creating-users/create-a-user#2-request-body) `nickname` is the **required** field. It means that if after all resolutions the `nickname` becomes `undefined` I think it is a good idea to throw an error, especially that we know beforehand that this field is indeed required by Sendbird.

To make things clear, even though we know Sendbird will throw an error without `nickname` do you want us to remove that error and let Sendbird throw an error for the user, correct?

Reply:
Ah ok I was missing that (I asked [here](https://github.com/awell-health/awell-extensions/pull/239#discussion_r1317036389) but received no confirmation) -- if that's the case then please alter the logic: 

- use nickname if present
- use firstName lastName if both present
- if either firstName or lastName is not present then use what you have (only firstName or lastName)
- if both are missing then throw an error as you have already (no change)